app/Jobs/TweetNewIssue.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Jobs;
+
+use App\DataTransferObjects\Issue;
+use App\Facades\Twitter;
+use App\Models\SocialPost;
+use Cache;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\Middleware\RateLimited;
+use Illuminate\Queue\SerializesModels;
+use RateLimiter;
+
+class TweetNewIssue implements ShouldQueue, ShouldBeUnique
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    public function __construct(protected Issue $issue)
+    {
+        //
+    }
+
+    public function handle(): void
+    {
+        Cache::lock('twitter-lock')->block(10, function () {
```

Comment:
How come we need this locking? Is it protecting us from anything in particular?

Reply:
If there are multiple workers, this prevents those two workers from sending a tweet at the same time. This might be covered by the rate limiter, but I'm not sure.